service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -547,18 +580,19 @@ func (t *transferQueueActiveTaskExecutor) processSignalExecution(
 			ctx,
 			task,
 			weContext,
-			targetNamespace,
+			targetNamespaceName,
 			task.TargetWorkflowID,
 			task.TargetRunID,
 			attributes.Control,
+			enumspb.SIGNAL_EXTERNAL_WORKFLOW_EXECUTION_FAILED_CAUSE_EXTERNAL_WORKFLOW_EXECUTION_NOT_FOUND,
```

Comment:
It's rare but namespace not found is also possible here I think. And currently we can't tell from the error if it's workflow not found or namespace not found.

Reply:
This is actually true for all 3 cases. I plan to add specific error for "namespace is not found" case and I will handle it here properly. Added `TODO` for now.